src/renderers/webgl/WebGLState.js

```diff
@@ -525,80 +525,123 @@ function WebGLState( gl, extensions, utils ) {
 
 	function setBlending( blending, blendEquation, blendSrc, blendDst, blendEquationAlpha, blendSrcAlpha, blendDstAlpha, premultipliedAlpha ) {
 
-		if ( blending !== NoBlending ) {
+		if ( blending === NoBlending ) {
 
-			enable( gl.BLEND );
+			if ( currentBlending !== NoBlending ) {
 
-		} else {
+				disable( gl.BLEND );
+				//blendingEnabled = false;
+				currentBlending = NoBlending;
+				console.log( 'disable blending' );
+
+			}
 
-			disable( gl.BLEND );
+			return;
 
 		}
 
-		if ( blending !== CustomBlending ) {
+		if ( currentBlending === NoBlending ) { // could test on blendingEnabled, instead
 
-			if ( blending !== currentBlending || premultipliedAlpha !== currentPremultipledAlpha ) {
+			enable( gl.BLEND ); // only enable if currently disabled
+			//blendingEnabled = true;
+			console.log( 'enable blending' );
 
-				switch ( blending ) {
+		}
 
-					case AdditiveBlending:
+		if ( blending === NormalBlending ) {
 
-						if ( premultipliedAlpha ) {
+			if ( currentBlending !== NormalBlending || premultipliedAlpha !== currentPremultipledAlpha ) {
 
-							gl.blendEquationSeparate( gl.FUNC_ADD, gl.FUNC_ADD );
-							gl.blendFuncSeparate( gl.ONE, gl.ONE, gl.ONE, gl.ONE );
+				gl.blendEquation( gl.FUNC_ADD ); // todo: only set if not
 
-						} else {
+				if ( premultipliedAlpha ) {
 
-							gl.blendEquation( gl.FUNC_ADD );
-							gl.blendFunc( gl.SRC_ALPHA, gl.ONE );
+					gl.blendFuncSeparate( gl.ONE, gl.ONE_MINUS_SRC_ALPHA, gl.ONE, gl.ONE_MINUS_SRC_ALPHA );
+					console.log( 'NormalBlending premultipliedAlpha' );
 
-						}
-						break;
+				} else {
 
-					case SubtractiveBlending:
+					gl.blendFuncSeparate( gl.SRC_ALPHA, gl.ONE_MINUS_SRC_ALPHA, gl.ONE, gl.ONE_MINUS_SRC_ALPHA );
+					console.log( 'NormalBlending' );
 
-						if ( premultipliedAlpha ) {
+				}
 
-							gl.blendEquationSeparate( gl.FUNC_ADD, gl.FUNC_ADD );
-							gl.blendFuncSeparate( gl.ZERO, gl.ZERO, gl.ONE_MINUS_SRC_COLOR, gl.ONE_MINUS_SRC_ALPHA );
+				currentBlending = NormalBlending;
+				currentPremultipledAlpha = premultipliedAlpha;
 
-						} else {
+			}
 
-							gl.blendEquation( gl.FUNC_ADD );
-							gl.blendFunc( gl.ZERO, gl.ONE_MINUS_SRC_COLOR );
+			return;
 
-						}
-						break;
+		}
+
+		console.log( 'special case blending' );
+
+		if ( blending !== CustomBlending ) {
+
+			if ( currentBlending !== blending || premultipliedAlpha !== currentPremultipledAlpha ) {
 
-					case MultiplyBlending:
+				gl.blendEquation( gl.FUNC_ADD );
 
-						if ( premultipliedAlpha ) {
+				if ( premultipliedAlpha ) {
+
+					switch ( blending ) {
+
+						case AdditiveBlending:
```

Comment:
Consider to write it a little bit more compact:
```js
switch ( blending ) {

   case AdditiveBlending:
      gl.blendFunc( gl.ONE, gl.ONE );
      break;

   case SubtractiveBlending:
      gl.blendFuncSeparate( gl.ZERO, gl.ZERO, gl.ONE_MINUS_SRC_COLOR, gl.ONE_MINUS_SRC_ALPHA );
      break;
```
Similar to:
https://github.com/mrdoob/three.js/blob/3585caf4f5779c5eb433edd2838257cd65113e33/src/renderers/webgl/WebGLProgram.js#L55-L63

Reply:
Done!